packages/vega-functions/src/functions/lasso.js

```diff
@@ -0,0 +1,105 @@
+import intersect from './intersect';
+
+
+/**
+ * Appends a new point to the lasso
+ * 
+ * @param {*} lasso the lasso in pixel space
+ * @param {*} x the x coordinate in pixel space
+ * @param {*} y the y coordinate in pixel space
+ * @param {*} minDist the minimum distance, in pixels, that thenew point needs to be apart from the last point
+ * @returns a new array containing the lasso with the new point
+ */
+export function lassoAppend(lasso, x, y, minDist = 5) {
+  const last = lasso[lasso.length - 1];
+
+  // Add point to lasso if distance to last point exceed minDist or its the first point
+  if (last === undefined || Math.sqrt(((last[0] - x) ** 2) + ((last[1] - y) ** 2)) > minDist) {
+    lasso.push([x, y]);
+
+    return [...lasso];
+  }
+
+  return lasso;
+}
+
+
+/**
+ * Generates a svg path command which draws a lasso
+ * 
+ * @param {*} lasso the lasso in pixel space in the form [[x,y], [x,y], ...]
+ * @returns the svg path command that draws the lasso
+ */
+export function lassoPath(lasso) {
+  return (lasso ?? []).reduce((svg, [x, y], i) => {
```

Comment:
using entries().reduce gave an error for me, this version works
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce